### ISSUE TITLE
Let people use an emoji response for `++` pile-ons

### ISSUE THREAD

**Issue by jcardinal** [jcardinal/hubot#19]
To reduce channel flooding and make it easier to give points, figure out and implement a method for letting people use an emoji response for giving points for the most recently-used reason.

**Comment by unsay** [comment]
There are two notable reaction events for +/- behaviors:

https://api.slack.com/events/reaction_added
https://api.slack.com/events/reaction_removed

The UX is going to be…curious. Maybe surrounding `+:emoji:+` with the signifier will make it worth points. Other ideas?

**Comment by jcardinal** [comment]
Ahh yes, this is complicated, because emoji reactions have a concept of undo, but hobo points don't. My thinking was to add a custom `++` emoji, and then award additional points when that reaction is used on a `thing++ for reason` post or a hobo response stating `thing`'s point total and how many of them are for the current `reason`.

I'm personally not interested in trying to get hobo to award points based on an emoji reaction to any and all posts; complicated to understand what we're awarding points to or what the reason would be. For example, if I post "unsay is the best for enabling this feature!" and then people use the `++` reaction on my post, how does hobo know whether to award the points to me, unsay, best, feature, etc? If we limit it to pile-ons after someone has intentionally and explicitly given points, I think that simplifies the UX.

As for undo, I think it's conceptually incompatible with how hobo points work, and it's probably fine to just accept that hobo won't undo anything if someone undoes their emoji reaction.

What do you think?

**Comment by unsay** [comment]
Good call…and in the end, it doesn't have to change much. The initial `hobo` response should include an emoji response e.g.

![image](https://user-images.githubusercontent.com/46175/46030034-a46f2e80-c0ba-11e8-813e-1c9bdd960814.png)

Where any `++` pile-ons are in the `hobo` response.

**Comment by jcardinal** [comment]
Ooh yes, I really like that!

A next step also just occurred to me: I'm sure the API allows for this, but I have no idea how difficult it would be...have hobo _edit_ its response message, updating the point total as people use the emoji response, instead of posting new replies to the channel.

Related: I'm trying to culturally encourage positive point giving, and discourage taking points away. If you're comfortable with that approach, I'd ask that you not enable a similar response for taking points away.
